decode_response.go

```diff
@@ -70,6 +72,13 @@ func (sp *SAMLServiceProvider) ValidateEncodedResponse(encodedResponse string) (
 	doc := etree.NewDocument()
 	err = doc.ReadFromBytes(raw)
 	if err != nil {
+		// Attempt to inflate the response in case it happens to be compressed (as with one case at saml.oktadev.com)
+		buf, flateErr := ioutil.ReadAll(flate.NewReader(bytes.NewReader(raw)))
+		if flateErr == nil {
+			err = doc.ReadFromBytes(buf)
+		}
+	}
+	if err != nil {
 		return nil, err
 	}
```

Comment:
I think this is missing a return in the case where `flateErr` is not nil.

I think it would be clearer if factored like this (haven't tested this though):

```go
 	doc := etree.NewDocument()
  	err = doc.ReadFromBytes(raw)
 	if err != nil {
 		// Attempt to inflate the response in case it happens to be compressed (as with one case at saml.oktadev.com)
 		buf, err := ioutil.ReadAll(flate.NewReader(bytes.NewReader(raw)))
 		if err != nil {
 		 	return nil, err
 		}

 		err = doc.ReadFromBytes(buf)
 		if err != nil {
 		 	return nil, err
 		}
 	}
```